core-common/src/main/java/org/glassfish/jersey/internal/config/SystemPropertiesConfigurationModel.java

```diff
@@ -0,0 +1,216 @@
+import javax.ws.rs.core.Feature;
+import java.lang.reflect.Field;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.logging.Logger;
+
+
+public class SystemPropertiesConfigurationModel implements ExternalConfigurationModel<Void> {
+
+    private static final Logger log = Logger.getLogger(SystemPropertiesConfigurationModel.class.getName());
+
+
+    private static final Map<Class, Function> converters = new HashMap<>();
+    static {
+        converters.put(String.class, (Function<String, String>) s -> s);
+        converters.put(Integer.class, (Function<String, Integer>) s -> Integer.valueOf(s));
+        converters.put(Boolean.class, (Function<String, Boolean>) s -> s.equalsIgnoreCase("1")
+                ? true
+                : Boolean.parseBoolean(s));
+    }
+
+    private String getSystemProperty(String name) {
+        return AccessController.doPrivileged(PropertiesHelper.getSystemProperty(name));
+    }
+
+    @Override
+    public <T> T as(String name, Class<T> clazz) {
+        if (converters.get(clazz) == null) {
+            throw new IllegalArgumentException("Unsupported class type");
+        }
+        return (name != null && clazz != null && isProperty(name))
+                ? clazz.cast(converters.get(clazz).apply(getSystemProperty(name)))
+                : null;
+    }
+
+
+
+    @Override
+    public <T> Optional<T> getOptionalProperty(String name, Class<T> clazz) {
+        return Optional.of(as(name, clazz));
+    }
+
+    @Override
+    public ExternalConfigurationModel mergeProperties(Map<String, Object> inputProperties) {
+        return this;
+    }
+
+    @Override
+    public Void getConfig() {
+        return null;
+    }
+
+    @Override
+    public boolean isProperty(String name) {
+        return Optional.ofNullable(
+                AccessController.doPrivileged(
+                        PropertiesHelper.getSystemProperty(name)
+                )
+        ).isPresent();
+    }
+
+    @Override
+    public RuntimeType getRuntimeType() {
+        return null;
+    }
+
+    @Override
+    public Map<String, Object> getProperties() {
+        final Map<String, Object> result = new HashMap<>();
+
+        final Boolean allowSystemPropertiesProvider = as(
+                CommonProperties.ALLOW_SYSTEM_PROPERTIES_PROVIDER, Boolean.class
+        );
+        if (!Boolean.TRUE.equals(allowSystemPropertiesProvider)) {
+            log.finer("System properties configuration provider not allowed");
+            return result;
+        }
+
+        try {
+            AccessController.doPrivileged(PropertiesHelper.getSystemProperties())
+                    .forEach((k, v) -> result.put(String.valueOf(k), v));
+        } catch (SecurityException se) {
+            log.warning(se.getLocalizedMessage());
```

Comment:
Please create a message that would make more clear in what case the warning arose. For instance "Could not read system properties:"